packages/core/src/plugins/rsdoctor.ts

```diff
@@ -45,9 +48,14 @@ export const pluginRsdoctor = (): RsbuildPlugin => ({
 
       let isAutoRegister = false;
 
+      function isRsdoctorPlugin(plugin: Configuration['plugins'] & { isRsdoctorPlugin?: boolean }) {
+        return plugin != null && typeof plugin.isRsdoctorPlugin === 'boolean';
+      }
+
       for (const config of bundlerConfigs) {
-        const registered = config.plugins?.some(
-          (plugin) => plugin?.constructor?.name === pluginName,
+        // If user has add Rsdoctor plugin in config file, will return.
+        const registered = config.plugins?.some(          
+          (plugin) => isRsdoctorPlugin(plugin as unknown as Configuration['plugins'] & { isRsdoctorPlugin?: boolean }) || plugin?.constructor?.name === pluginName,
```

Comment:
We can move the `plugin?.constructor?.name === pluginName` to `isRsdoctorPlugin()` helper.

Reply:
> We can move the `plugin?.constructor?.name === pluginName` to `isRsdoctorPlugin()` helper.

@chenjiahan I just worry about remove the `plugin?.constructor?.name === pluginName` may be a break change.

Reply:
No, I mean move the logic to the function, not to remove it.